drivers/ccs811/ccs811.c

```diff
@@ -0,0 +1,619 @@
+     */
+    if (!(status & CCS811_STATUS_FW_MODE))
+    {
+        /* check whether valid application firmware is loaded */
+        if (!(status & CCS811_STATUS_APP_VALID)) {
+            DEBUG_DEV("sensor is in boot mode, but has no app", dev);
+            return -CCS811_ERROR_NO_APP;
+        }
+
+        /* swtich to application mode */
+        if (_reg_write(dev, CCS811_REG_APP_START, 0, 0) != CCS811_OK) {
+            DEBUG_DEV("could not write app start command "
+                      "to register CCS811_REG_APP_START", dev);
+            return -CCS811_ERROR_I2C;
+        }
+
+        /* wait 100 ms after starting the app */
+        xtimer_usleep (100000);
+
+        /* get the status to check whether sensor switched to application mode */
+        if (_reg_read(dev, CCS811_REG_STATUS, &status, 1) != CCS811_OK) {
+            DEBUG_DEV("could not read register CCS811_REG_STATUS", dev);
+            return -CCS811_ERROR_I2C;
+        }
+        if ((status & CCS811_STATUS_FW_MODE) == 0) {
+            DEBUG_DEV("could not start application", dev);
+            return -CCS811_ERROR_NO_APP;
+        }
+    }
+
+    /* try to set default measurement mode */
+    return ccs811_set_mode(dev, dev->mode);
+}
+
+
+int ccs811_set_mode (ccs811_dev_t* dev, ccs811_mode_t mode)
+{
+    ASSERT_PARAM (dev != NULL);
+
+    ccs811_meas_mode_reg_t reg;
+
+    /* read measurement mode register value */
+    if (_reg_read(dev, CCS811_REG_MEAS_MODE, (uint8_t*)&reg, 1) != CCS811_OK) {
+        DEBUG_DEV("could not read current measurement mode "
+                  "from register CCS811_REG_MEAS_MODE", dev);
+        return -CCS811_ERROR_I2C;
+    }
+
+    reg.drive_mode = mode;
+
+    /* write back measurement mode register */
+    if (_reg_write(dev, CCS811_REG_MEAS_MODE, (uint8_t*)&reg, 1) != CCS811_OK) {
+        DEBUG_DEV("could not write new measurement mode "
+                  "to register CCS811_REG_MEAS_MODE", dev);
+        return -CCS811_ERROR_I2C;
+    }
+
+    /* check whether setting measurement mode were succesfull */
+    if (_reg_read(dev, CCS811_REG_MEAS_MODE, (uint8_t*)&reg, 1) != CCS811_OK) {
+        DEBUG_DEV("could not read new measurement mode "
+                  "from register CCS811_REG_MEAS_MODE", dev);
+        return -CCS811_ERROR_I2C;
+    }
+    if (reg.drive_mode != mode) {
+        DEBUG_DEV("could not set measurement mode to %d", dev, mode);
+        return -CCS811_ERROR_MEASMODE_INV;
+    }
+
+    dev->mode = mode;
+
+    return CCS811_OK;
+}
+
+
+int ccs811_data_ready (ccs811_dev_t* dev)
+{
+    uint8_t status;
+
+    /* check status register */
+    if (_reg_read(dev, CCS811_REG_STATUS, &status, 1) != CCS811_OK) {
+        DEBUG_DEV ("could not read CCS811_REG_STATUS", dev);
+        return -CCS811_ERROR_I2C;
+    }
+
+    if ((status & CCS811_STATUS_DATA_RDY)) {
+        /* new data available */
+        return CCS811_OK;
+    }
+
+    return CCS811_ERROR_NO_NEW_DATA;
+}
+
+
+#define CCS811_ALG_DATA_ECO2_HB   0
```

Comment:
@aabadie @haukepetersen Your suggestion was to place such internal driver stuff in a header file called ```ccs811_internal.h```. However, when this file is located in ```drivers/ccs811/include```, as I have seen for other drivers, it is not really internal since it is visible also from other directories because of the INCLUDES path and can be included by arbitrary C files. It would be internal and only visible for ```ccs811.c``` if is located in the same directory as the ```ccs811.c``` file.

Why not to leave the internal driver stuff with local scope in the file where they should be used exclusively, as with static variables and functions? That is not really clear to me. Usually, defines and declarations are placed in header files when they are part of an interface and have to be used by multiple C files. I just wanna to avoid naming conflicts with other header files.

Following the ```l3g4200d``` driver as example, I have now a file ```drivers/ccs811/include/ccs811_regs.h``` but as the name suggests, only register definitions should be placed there.